projects/storefrontlib/src/events/product/product-page-event.builder.ts

```diff
@@ -72,23 +72,20 @@ export class ProductPageEventBuilder {
       skip(1)
     );
 
-    const categoryPageEvent$ = this.eventService.get(PageEvent).pipe(
-      map((pageEvent) => ({
-        isCategoryPage: pageEvent.semanticRoute === 'category',
-        categoryCode: pageEvent.context.id,
-      }))
-    );
-
-    return categoryPageEvent$.pipe(
+    return this.eventService.get(PageEvent).pipe(
       switchMap((pageEvent) => {
-        if (!pageEvent.isCategoryPage) {
+        if (pageEvent?.semanticRoute !== 'category') {
```

Comment:
any reason to not filter?

Reply:
Yes, it breaks the stream. The idea is to _always_ switch from the base `PageEvent` and then further inspect the event and conditionally return an appropriate stream (`EMPTY` or the actual event stream)